flytekit/common/tasks/sdk_dynamic.py

```diff
@@ -150,10 +157,28 @@ def _produce_dynamic_job_spec(self, context, inputs):
         visited_nodes = set()
         generated_ids = {}
         effective_failure_ratio = self._allowed_failure_ratio or 0.0
+
+        # TODO: This function needs to be cleaned up.
+        # The reason we chain these two together is because we allow users to not have to explicitly "yield" the
+        # node. As long as the subtask/lp/subwf has an output that's referenced, it'll get picked up.
         for sub_task_node in _itertools.chain(yielded_sub_tasks, upstream_nodes):
             if sub_task_node in visited_nodes:
                 continue
             visited_nodes.add(sub_task_node)
+            executable = sub_task_node.executable_sdk_object
+
+            # If the executable object that we're dealing with is registerable (ie, SdkRunnableLaunchPlan, SdkWorkflow
+            # SdkTask, or SdkRunnableTask), then it should have the ability to give itself a name. After assigning
+            # itself the name, also make sure the id is properly set according to current config values.
+            if isinstance(executable, _registerable.RegisterableEntity):
+                executable.auto_assign_name()
+                executable._id = _identifier.Identifier(
+                    executable.resource_type,
+                    _internal_config.TASK_PROJECT.get() or _internal_config.PROJECT.get(),
+                    _internal_config.TASK_DOMAIN.get() or _internal_config.DOMAIN.get(),
+                    executable._platform_valid_name,
```

Comment:
we should make this a property on the RegisterableEntity object so that we don't read access a 'private' attribute. Someday I'll figure out the right pattern so we don't have to do this `executable._id` stuff too :(